src/page.rs

```diff
@@ -303,6 +303,65 @@ impl Page {
         Ok(self)
     }
 
+    /// Wait for the network to be idle for 500ms
+    #[cfg(feature = "tokio-runtime")]
+    pub async fn wait_for_network_idle(&self) -> Result<&Self> {
+        let mut events = self.event_listener::<chromiumoxide_cdp::cdp::browser_protocol::network::EventLoadingFinished>().await?;
+
+        loop {
+            let sleep = tokio::time::sleep(tokio::time::Duration::from_millis(500));
+            tokio::pin!(sleep);
+            tokio::select! {
+                _ = &mut sleep => break,
+                _ = events.next() => (),
+                else => break,
+            }
```

Comment:
hmm, how many events do we expect here?

so this waits for the first event plus an additional 500ms? at most 1s

if we only expect 1 event then the additional timeout( wrapper isn't useful

Reply:
It allows (x) events until no events are found for 500ms. The timeout wrapper protects from situations when a website sends repeated events in a loop etc.